examples/hosts/app-integration/schema-upgrade/src/containerKillBit.ts

```diff
@@ -0,0 +1,120 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TaskManager } from "@fluid-experimental/task-manager";
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { IEvent, IEventProvider } from "@fluidframework/common-definitions";
+import { ConsensusRegisterCollection } from "@fluidframework/register-collection";
+// import { IFluidHandle } from "@fluidframework/core-interfaces";
+
+const crcKey = "crc";
+const taskManagerKey = "task-manager";
+const markedForDestructionKey = "marked";
+const destroyTaskName = "destroy";
+const deadKey = "dead";
+
+export interface IContainerKillBitEvents extends IEvent {
+    (event: "markedForDestruction" | "dead", listener: () => void);
+}
+
+export interface IContainerKillBit extends IEventProvider<IContainerKillBitEvents> {
+    dead: boolean;
+    setDead(): Promise<void>;
+    markedForDestruction: boolean;
+    markForDestruction(): Promise<void>;
+    volunteerForDestruction(): Promise<void>;
+    haveDestructionTask(): boolean;
+}
+
+export class ContainerKillBit extends DataObject implements IContainerKillBit {
+    private _crc: ConsensusRegisterCollection<boolean> | undefined;
+    private _taskManager: TaskManager | undefined;
+
+    private get crc() {
+        if (this._crc === undefined) {
+            throw new Error("Couldn't retrieve the ConsensusRegisterCollection");
+        }
+        return this._crc;
+    }
+
+    private get taskManager() {
+        if (this._taskManager === undefined) {
+            throw new Error("Couldn't retrieve the TaskManager");
+        }
+        return this._taskManager;
+    }
+
+    public get dead() {
+        return this.crc.read(deadKey) as boolean;
+    }
+
+    public async setDead() {
+        // Using a consensus-type data structure here, to make it easier to validate
+        // that the setDead was ack'd and we can have confidence other clients will agree.
+        await this.crc.write(deadKey, true);
+    }
+
+    public get markedForDestruction() {
+        return this.crc.read(markedForDestructionKey) as boolean;
+    }
+
+    public async markForDestruction() {
+        // This should probably use a quorum-type data structure here.
+        // Then, when everyone sees the quorum proposal get approved they can choose to either volunteer
+        // or close themselves
+        await this.crc.write(markedForDestructionKey, true);
+    }
+
+    public async volunteerForDestruction(): Promise<void> {
+        return this.taskManager.lockTask(destroyTaskName);
```

Comment:
This will throw on disconnect. Is this expected by callers?

Reply:
One note - this demo is not in its final state, but rather just a baseline we will iterate upon (this starting point was my external data proof of concept from last fall).  So it's not yet trying to be airtight against the full variety of scenarios, though eventually it should be.

@alpaix perhaps we should add some notes in the README.md about the status of this demo and current known gaps to set expectations.

Reply:
Noted to handle disconnect scenario properly in #9717